### PR TITLE
Reduce Bottom Spacing for Seamless Panels

### LINKED ISSUE #2222: Reduce vertical space usage at the bottom of panels
v4.1.0

Current:

image

Suggestion: It is good if we can reduce this empty space. e.g.,

image

### PR DESCRIPTION
**Pull request by EssWhyy**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [x] Others, please explain:

Fixes #2222

**Overview of changes:**
I have modified the CSS modifiers for the panel cards button and bottom hr tags to reduce the empty space at the bottom of the panels. Intended effect is shown below. 

possible panel css fix

**Anything you'd like to highlight/discuss:**
I am not certain if the hr tag css I changed in NestedPanel.vue is the correct css. Since bootstrap.min.css also has a css for hr tags, but I'm not allowed to change it or it will fail tests.   

**Testing instructions:**
NA

**Proposed commit message: (wrap lines at 72 characters)**

Reduce margin spacing for hr tag within seamless panels

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes

### CODE REVIEW

**Comment by tlylt** [comment]
Thanks, @EssWhyy for tackling this.

> image

1. Image is broken

> I am not certain if the hr tag css I changed in NestedPanel.vue is the correct css. 

I think it sounds reasonable to change the CSS in NestedPanel, but any reason why `!important` is used and are there possible alternatives? An explanation for this would be useful as using !important unnecessarily might have unintended consequences.

> Since bootstrap.min.css also has a css for hr tags, but I'm not allowed to change it or it will fail tests.

Do you mean you edit the min.css file directly? what changes did you make and what tests are failing?

**Comment by EssWhyy** [comment]
Hello @tlylt apologies for the late reply, here is the screenshot I meant to send. 

possible panel css fix

The use of !important tag was to override the bootstrap css for the hr tags. But I decided to remove them since the panel css should override it by default.

I initially tried to edit some hr css in markbind\packages\core-web\asset\css\bootstrap.min.css, but this leads to the markbind-cli:test failing. This probably is also not the best way to approach it since it affects all other hr tags.

There does not seem to be any panel in the user guide docs with a hr element, unlike the example in the issue which appears to be taken from the CS2103T website. Is there a specific class for the panels in CS2103T with the bottom hr lines?

**Comment by tlylt** [comment]
> I initially tried to edit some hr css in markbind\packages\core-web\asset\css\bootstrap.min.css, but this leads to the markbind-cli:test failing. This probably is also not the best way to approach it since it affects all other hr tags.

yes I don't think you will ever need to edit the min.css file, unless upgrading bootstrap itself.

> There does not seem to be any panel in the user guide docs with a hr element, unlike the example in the issue which appears to be taken from the CS2103T website. Is there a specific class for the panels in CS2103T with the bottom hr lines?

Is this panel what you refer to as a panel with a hr element?
image

Besides the above, in the panel section, the distance between the button and the closing bottom line has been impacted. Is this intended by you? This UI change seems to not look as good when it's uniform across all panels.
image

Compare the before https://markbind-master.netlify.app/userguide/components/presentation#panels and after https://deploy-preview-2319--markbind-master.netlify.app/userguide/components/presentation#panels when doing your testing.

**Comment by damithc** [comment]
@EssWhyy minor: can you update the PR title to be a more descriptive one?

**Comment by EssWhyy** [comment]
For the latest commit I reverted my previous changes and added a new CSS selector within NestedPanel.vue to adjust the margins of the seamless panel hr directly. The effect is shown below.

gap change

File: packages/vue-components/src/panels/NestedPanel.vue [hunk 1]
```diff
@@ -358,7 +363,7 @@ export default {
         }
 
         .card-collapse > hr {
-            margin-top: 1.5rem;
+            margin-top: 1rem;
```
Comment (tlylt) [inline packages/vue-components/src/panels/NestedPanel.vue:366]:
```suggestion
            margin-top: 0.5rem;
```
This seems to achieve the same effect for larger screen. With 1rem:
image
But with 0.5rem:
image

File: packages/vue-components/src/panels/NestedPanel.vue [hunk 2]
```diff
@@ -275,6 +275,11 @@ export default {
         margin-top: 5px;
     }
 
+    .expandable-card > .card-collapse > hr {
+        margin-top: 0.25rem;
+        margin-bottom: 0.75rem;
```
Comment (tlylt) [inline packages/vue-components/src/panels/NestedPanel.vue:280]:
```suggestion
```
Any reason why `margin-bottom` is adjusted? Doesn't seem to play any role in achieving the outcome

**Review by tlylt — CHANGES_REQUESTED** [review]
Thanks @EssWhyy for following up on this. Review comments given.

Reference links:
- the page that contains the panel mentioned in the issue: https://nus-cs2103-ay2223s2.github.io/website/admin/tp-expectations.html (Expectation Do an equal share of the team-tasks.)
- some seamless panel in our UG: https://markbind-master.netlify.app/index.html (More examples of generating static content)

**Review by tlylt — APPROVED** [review]

**Comment by tlylt** [comment]
@all-contributors please add @EssWhyy for code

**Pull request merged**